### PR TITLE
chore: bump eslint dependencies

### PR DESCRIPTION
**Pull request by bradzacher**
## PR Checklist

- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Bumps our dev dep dependencies on ESLint packages across the board to the latest.
I could have just limited this to the official eslint packages but I figured "why not do everything".

Note: I made the executive decision to remove `eslint-plugin-sonarjs` here. They pin all of their deps to specific versions and also ship so many deps it ended up introducing tens of duplicate versions into our repo.
screenshot of sonars deps
Simply put: the weight of it was not worth the one rule we used from it.
Removing this removed ~700 lines of changes from the `yarn.lock`.

### CODE REVIEW

**Review by kirkwaiblinger — APPROVED** [review]

**Comment by bradzacher** [comment]
gonna merge with 1 stamp just cos i want to land this before the release to prove to people we fixed the 9.15 regression
